python/pyspark/mllib/linalg.py

```diff
@@ -821,6 +821,23 @@ def __reduce__(self):
             self.numRows, self.numCols, self.values.tostring(),
             int(self.isTransposed))
 
+    def __str__(self):
+        mattoarr = self.toArray()
+        matstr = ""
+        for row in mattoarr:
+            for ind, col in enumerate(row):
+                if ind != 0:
+                    matstr += "  "
+                matstr += str(col)
+            matstr += '\n'
```

Comment:
This will not work since row is not a string. But using the same logic. I've updated it in a better way,